lib/ddtrace/configuration/settings.rb

```diff
@@ -42,19 +44,44 @@ def initialize(*_)
         tags
       end
 
+      # Legacy [App Analytics](https://docs.datadoghq.com/tracing/legacy_app_analytics/) configuration.
+      #
+      # @deprecated Use [Trace Retention and Ingestion](https://docs.datadoghq.com/tracing/trace_retention_and_ingestion/)
+      #   controls.
+      # @public_api
       settings :analytics do
+        # @default `DD_TRACE_ANALYTICS_ENABLED` environment variable, otherwise `nil`
+        # @return [Boolean,nil]
         option :enabled do |o|
           o.default { env_to_bool(Ext::Analytics::ENV_TRACE_ANALYTICS_ENABLED, nil) }
           o.lazy
         end
       end
 
+      # Profiler API key.
+      # @default `DD_API_KEY` environment variable, otherwise `nil`
+      # @return [String,nil]
+      # @public_api
       option :api_key do |o|
```

Comment:
This one is a though one, since it's not really just for the profiler (other datadog producs use `DD_API_KEY`), it's just that only the profiler /right now/ makes use of it.

I suggest marking it as "For internal use only", even if it's still public (maybe...?).

Reply:
Should we have the option, but not make it part of the public API then, given your rationale?
I don't think I grasp 100% the situation here.

Reply:
> I don't think I grasp 100% the situation here.

Yup, sorry for the confusion.

My thinking is -- we want to keep these options. With that in mind, I think it doesn't quite make sense to have "private options". But documentation-wise, we still want to steer customers away from them.

Something like this (lol):

![](https://previews.123rf.com/images/dondesigns/dondesigns1409/dondesigns140900005/31501377-do-not-press-push-button-isolated-on-white-background.jpg)

So my proposal is that for site and api_key we:

1. Don't mark it as `@public_api`
2. Don't explain what it is (or at least, an explanation of what it is should not appear in the generated docs -- an internal comment is fine)
3. Just state that this is "For internal Datadog testing use. Changing it is not recommended".